### PR TITLE
refactor(booking-agent): migrate whatsapp agent to langgraph booking-agent architecture

### PR DESCRIPTION
**Pull request by dcodesmith**
This change migrates the conversational booking flow to a booking-agent architecture powered by LangGraph, replacing legacy WhatsApp-specific orchestration and introducing a more deterministic, testable state machine for extraction, routing, search, booking creation, and response delivery.
1) Agent architecture and module migration
Renames/restructures the old whatsapp-agent domain into a medium-agnostic booking-agent module, with WhatsApp-specific transport concerns isolated under booking-agent/whatsapp.
Removes legacy orchestration/search-memory/follow-up code paths and feature-flag-era branching, reducing duplicated decision logic and simplifying runtime behavior.
Adds explicit booking-agent interfaces, constants, and error types to enforce stable contracts between graph nodes, services, and transport adapters.
2) LangGraph-driven booking workflow
Introduces a full LangGraph pipeline:
extract user intent and slot patches from text/interactive inputs
merge into persisted draft state
route to search, confirm, handoff, cancel/reset, or booking creation
respond with deterministic or model-assisted messaging
outbox build for transport-safe message dispatch
Improves routing determinism for critical control intents (confirm/reject/agent/cancel/reset), reducing misroutes caused by model drift.
Adds clearer stage transitions (collecting, presenting_options, confirming, awaiting_payment, etc.) with guarded fallback behavior.
3) Vehicle search behavior and option handling
Consolidates search logic into booking-agent-search.service with precondition checks and alternative ranking.
Keeps exact-match-first behavior while supporting ranked alternatives when exact vehicles are unavailable.
Strengthens selection resolution so users can choose by ordinal, price preference (e.g., cheapest), id, make/model/color hints.
4) Booking creation reliability and user-safe failures
Enhances booking creation flow to prevent raw internal errors from leaking to users.
Handles car unavailability during confirmation by recovering into refreshed option presentation where possible, instead of hard-failing the conversation.
Adjusts validation-sensitive payload behavior to avoid avoidable booking mismatches.
Improves booking summary rendering (including duration visibility).
5) Outbound delivery and WhatsApp channel hardening
Adds a dedicated persistence service for WhatsApp conversation/message/outbox storage responsibilities.
Improves outbox processing semantics and retries, with explicit status transitions and dead-letter behavior.
Keeps queue-driven processing boundaries clear between ingress, processor, orchestrator, and sender.
Adds voice-note transcription support in the inbound processor path so audio can be converted to text before orchestration.
6) OpenAI client consistency via DI
Introduces shared OPENAI_SDK_CLIENT provider module for SDK-based features.
Refactors AI consumers (including ai-search and audio transcription) to use DI rather than local ad-hoc client construction.
Keeps LangChain ChatOpenAI usage separate from raw SDK usage, while sharing the same env-backed API key source.
7) Checkout CTA template functionality (latest stage behavior)
Implements awaiting-payment checkout delivery using a Twilio template-compatible variable flow.
Extracts checkout token from full Flutterwave URL (.../pay/{token}) for URL-variable template usage.
Supports template variable mapping for body + token ({{1}} body text, {{2}} token), including selected vehicle make/model in body text.
Adds graceful fallback to freeform link delivery when token extraction fails, ensuring payment flow remains unblocked.
8) Test coverage and confidence
Adds/updates extensive unit tests across graph policy/services, outbox builder, WhatsApp runtime services, and OpenAI DI consumers.
Includes tests for:
stage routing and control intent guards
booking failure sanitization/recovery
template outbox payload construction
voice transcription and fallback behavior
queue/persistence edge cases

---

> [!NOTE]
> **High Risk**
> Large refactor of the WhatsApp booking conversation pipeline, including new LangGraph state machine orchestration, Redis-backed state, and booking creation/error-handling paths; regressions could impact customer messaging and booking/payment flows.
> 
> **Overview**
> Introduces a new **LangGraph-powered** `BookingAgentModule` and wires it into the app, replacing the previous `WhatsAppAgentModule` and moving WhatsApp transport concerns under `booking-agent/whatsapp`.
> 
> Adds a LangGraph state machine (`LangGraphGraphService`) with extraction/merge/routing/search/booking-creation/response nodes, plus deterministic control-intent + interactive-reply handling and safer booking-creation failure recovery (e.g., refresh options on car unavailability, sanitize unexpected errors).
> 
> Refactors vehicle search into `BookingAgentSearchService` (shared booking leg-count helper, enforced preconditions, alternative ranking, timeout handling) and updates AI-search OpenAI usage to a shared DI-provided SDK client with typed timeout/auth error mapping.
> 
> Updates configuration and dependencies for the new agent stack: adds `ANTHROPIC_API_KEY` + LangGraph env settings/fixtures, brings in LangChain/LangGraph packages, and pins `multer` to `2.1.0` via overrides/lockfile.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 73b2fbe261fcc28fcdc6e9d9e141fa8ab0161e56. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by dcodesmith** [comment]
@cursor review

**Comment by dcodesmith** [comment]
@cursor review

File: src/modules/booking-agent/langgraph/prompts/extractor.prompt.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import { addDays, format } from "date-fns";
+import type { BuildExtractorPromptInput } from "../langgraph.interface";
+
+export function buildExtractorSystemPrompt(input: BuildExtractorPromptInput): string {
+  const { currentDraft, lastShownOptions, stage, messages } = input;
+  const now = new Date();
+  const todayFormatted = format(now, "yyyy-MM-dd");
+  const timeFormatted = format(now, "HH:mm");
+  const tomorrowFormatted = format(addDays(now, 1), "yyyy-MM-dd");
+
```
[the remark this answers is not in the document]
Comment (dcodesmith) [reply src/modules/booking-agent/langgraph/prompts/extractor.prompt.ts:10]: Timezone is set https://github.com/dcodesmith/hyre-worker-nestjs/blob/884f462a7655b38367bbcb2c2c3958dfbffbf62a/src/main.ts#L84. Do we need to set it again?

Which is set as `TZ=Africa/Lagos` in the environment variables. Do we still need to do it where you suggested?

**Comment by dcodesmith** [comment]
@cursor review
